vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -574,36 +504,49 @@ static void add_paths(int start_node_ind,
             seg_index,
             box_id,
             delta};
-        CompressedRoutingCostKey compressed_key(key);
 
-        float new_delay = current_full.delay - parent_entry.delay;
-        float new_congestion = current_full.congestion_upstream - parent_entry.congestion_upstream;
+        if (*it != start_node_ind) {
+            auto& parent_node = device_ctx.rr_nodes[parent];
+            start_to_here = Entry(*it, parent_node.edge_switch((*paths)[*it].edge), &start_to_here);
+            parent = *it;
+        }
 
-        VTR_ASSERT(new_delay >= 0.f);
-        VTR_ASSERT(new_congestion >= 0.f);
+        float cost = current.cost() - start_to_here.cost();
+        if (cost < 0.f && cost > -10e-15 /* 10 femtosecond */) {
+            cost = 0.f;
+        }
 
-        auto val = util::Cost_Entry(new_delay, new_congestion);
+        VTR_ASSERT(cost >= 0.f);
 
         // NOTE: implements REPRESENTATIVE_ENTRY_METHOD == SMALLEST
-        auto result = routing_costs->insert(std::make_pair(key, val));
+        auto result = routing_costs->insert(std::make_pair(key, cost));
         if (!result.second) {
-            assign_min_entry(&result.first->second, val);
+            if (cost < result.first->second) {
+                result.first->second = cost;
+                new_sample_found = true;
+            } else if (BREAK_ON_MISS) {
+                break;
+            }
+        } else {
+            new_sample_found = true;
         }
-
-        // update parent data
-        parent_entry = util::PQ_Entry(*it, parent_node.edge_switch((*paths)[*it].edge), parent_entry.delay,
-                                      parent_entry.R_upstream, parent_entry.congestion_upstream, false, /*Tsw_adjust=*/0.f);
-        parent = *it;
     }
+    return new_sample_found;
 }
 
 /* runs Dijkstra's algorithm from specified node until all nodes have been
  * visited. Each time a pin is visited, the delay/congestion information
  * to that pin is stored to an entry in the routing_cost_map */
 template<typename Entry>
-static void run_dijkstra(int start_node_ind,
-                         RoutingCosts* routing_costs) {
+static std::pair<float, int> run_dijkstra(int start_node_ind,
```

Comment:
Need documentation on the return value.  Consider making them output arguments so they have descriptive names?

Reply:
Added documentation.